### PR TITLE
refactor rest client

### PR DESCRIPTION
**Pull request by TheMattSchiller**

### CODE REVIEW

File: client/restclient/restclient.go [hunk 1]
```diff
@@ -1,36 +1,43 @@
...
-type RestClient struct {
+// Client is the struct used interact with the Banyan REST API.
+type Client struct {
 	accessToken string
 	hostUrl     string
 	httpClient  *http.Client
 }
 
-const defaultHostUrl = "http://net.banyanops.com"
+const defaultHostUrl = "https://net.banyanops.com"
```
Comment (danielboros) [inline client/restclient/restclient.go:24]: We have a similar `const` elsewhere in this PR. Why?

File: client/restclient/restclient.go [hunk 2]
```diff
@@ -1,36 +1,43 @@
...
-// As part of this it exchanges the given refreshtoken for an acesstoken.
-func New(hostUrl string, refreshToken string, apiToken string) (client *RestClient, err error) {
+// As part of this it exchanges the given refreshtoken or api key
+func New(hostUrl string, refreshToken string, apiToken string) (client *Client, err error) {
 	clientHostUrl := defaultHostUrl
 	if hostUrl != "" {
 		clientHostUrl = hostUrl
 	}
 
+	if !strings.HasSuffix(hostUrl, "/") {
```
Comment (danielboros) [inline client/restclient/restclient.go:34]: What's the difference between this `Client` type and the one above in this PR?

File: client/restclient/restclient.go [hunk 3]
```diff
@@ -163,3 +171,119 @@ func getAccessTokenFromJSON(body []byte) (accessToken string, err error) {
...
+	return
+}
+
+func HandleResponse(response *http.Response, component string) (responseData []byte, err error) {
+	defer response.Body.Close()
+	responseData, err = ioutil.ReadAll(response.Body)
+	if err != nil {
+		return
+	}
+	if response.StatusCode == 404 || response.StatusCode == 400 {
```
Comment (danielboros) [inline client/restclient/restclient.go:273]: `http.StatusBadRequest` is not found? Seems like this could cause a bug.

**Review by danielboros — CHANGES_REQUESTED** [review]

File: client/restclient/restclient.go (hunk 2 above)
Comment (TheMattSchiller) [reply client/restclient/restclient.go:34]: apparently nothing. I removed it

**Review by danielboros — APPROVED** [review]

**Pull request merged**